pandas/_libs/reduction.pyx

```diff
@@ -275,6 +272,11 @@ cdef class SeriesGrouper(_BaseGrouper):
                     islider.move(start, end)
                     vslider.move(start, end)
 
+                    if cached_index is None:
```

Comment:
why can't this be done outside the loop?

Reply:
Because it seems the first call of `islider.move()` and `vslider.move()` is needed before initializing the cache (which use slider `buf`attribute), otherwise I get a series where size of index doesn't match size of values (see the non regression test).

I'm new to pandas code and didn't go in deep in this bug. It might be a better way to fix this, so expert eyes are welcome :) Especially I didn't not yet understand why this only occur with DatetimeIndex using UTC timezone (naive datetime don't trigger the bug).

Reply:
I'll take a closer look.  This file is pretty tricky.